src/plugins/share/public/components/tabs/link/index.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useCallback } from 'react';
+import { i18n } from '@kbn/i18n';
+import { copyToClipboard } from '@elastic/eui';
+import { type IModalTabDeclaration } from '@kbn/shared-ux-tabbed-modal';
+import { useShareTabsContext } from '../../context';
+import { LinkContent } from './link_content';
+
+type ILinkTab = IModalTabDeclaration<{
+  dashboardUrl: string;
+}>;
+
+const LINK_TAB_ACTIONS = {
+  SET_DASHBOARD_URL: 'SET_DASHBOARD_URL',
+};
+
+const linkTabReducer: ILinkTab['reducer'] = (
+  state = {
+    dashboardUrl: '',
+  },
+  action
+) => {
+  switch (action.type) {
+    case LINK_TAB_ACTIONS.SET_DASHBOARD_URL:
+      return {
+        ...state,
+        dashboardUrl: action.payload,
+      };
+    default:
+      return state;
+  }
+};
+
+const LinkTabContent: ILinkTab['content'] = ({ state, dispatch }) => {
+  const {
+    objectType,
+    objectId,
+    isDirty,
+    isEmbedded,
+    shareableUrl,
+    shareableUrlForSavedObject,
+    urlService,
+    shareableUrlLocatorParams,
+  } = useShareTabsContext()!;
+
+  const setDashboardLink = useCallback(
```

Comment:
would we add here the logic for the color of the buttons for Lens to show unsaved state like in the Figma? https://www.figma.com/file/PLgOUGbKgOPo8X6DVfUlQa/Share-modal?type=design&node-id=5-2216&mode=design&t=yPpDeEoSv9NhmrTd-0